backend/dal/Services/Concrete/BuildingService.cs

```diff
@@ -0,0 +1,266 @@
+            query = this.Context.Buildings
+                .Include(b => b.Parcel)
+                .AsNoTracking()
+                .Where(p =>
+                (!p.IsSensitive || (viewSensitive && userAgencies.Contains(p.AgencyId))) &&
+                p.Latitude != 0 &&
+                p.Longitude != 0 &&
+                p.Latitude <= neLat &&
+                p.Latitude >= swLat &&
+                p.Longitude <= neLong &&
+                p.Longitude >= swLong);
+            return query.ToArray();
+        }
+
+        /// <summary>
+        /// Get an array of buildings within the specified filter.
+        /// Will not return sensitive buildings unless the user has the `sensitive-view` claim and belongs to the owning agency.
+        /// </summary>
+        /// <param name="filter"></param>
+        /// <returns></returns>
+        public IEnumerable<Building> GetNoTracking(BuildingFilter filter)
+        {
+            filter.ThrowIfNull(nameof(filter));
+
+            // Check if user has the ability to view sensitive properties.
+            var userAgencies = this.User.GetAgencies();
+            var viewSensitive = this.User.HasPermission(Security.Permissions.SensitiveView);
+
+            // Users may only view sensitive properties if they have the `sensitive-view` claim and belong to the owning agency.
+            var query = this.Context.Buildings.AsNoTracking().Where(p =>
+                !p.IsSensitive || (viewSensitive && userAgencies.Contains(p.AgencyId)));
+
+            if (filter.NELatitude.HasValue && filter.NELongitude.HasValue && filter.SWLatitude.HasValue && filter.SWLongitude.HasValue)
+                query = query.Where(p =>
+                    p.Latitude != 0 &&
+                    p.Longitude != 0 &&
+                    p.Latitude <= filter.NELatitude &&
+                    p.Latitude >= filter.SWLatitude &&
+                    p.Longitude <= filter.NELongitude &&
+                    p.Longitude >= filter.SWLongitude);
+
+            if (filter.Agencies?.Any() == true)
+                query = query.Where(p => filter.Agencies.Contains(p.AgencyId));
+            if (filter.ConstructionTypeId.HasValue)
+                query = query.Where(p => p.BuildingConstructionTypeId == filter.ConstructionTypeId);
+            if (filter.PredominateUseId.HasValue)
+                query = query.Where(p => p.BuildingPredominateUseId == filter.PredominateUseId);
+            if (filter.FloorCount.HasValue)
+                query = query.Where(p => p.BuildingFloorCount == filter.FloorCount);
+            if (!String.IsNullOrWhiteSpace(filter.Tenancy))
+                query = query.Where(p => p.BuildingTenancy == filter.Tenancy);
+
+            if (!String.IsNullOrWhiteSpace(filter.Address)) // TODO: Parse the address information by City, Postal, etc.
+                query.Where(p => EF.Functions.Like(p.Address.Address1, $"%{filter.Address}%"));
```

Comment:
Did you forget to assign **query** or was this intentional? 
Shouldn't it be `query = query.Where(...)`?